airflow/www/views.py

```diff
@@ -2828,7 +2828,7 @@ def _convert_to_date(session, column):
             restriction = TimeRestriction(dag.start_date, dag.end_date, False)
             dates = collections.Counter()
 
-            if isinstance(dag.timetable, CronDataIntervalTimetable):
+            if isinstance(dag.timetable, CronTriggerTimetable):
```

Comment:
Hmm, actually I think this was intended for CronDataIntervalTimetable, since the code was initially an optimisation to make rendering faster, and actually predates CronTriggerTimetable.

So I think we should do two things

1. _Add_ CronTriggerTimetable to the check (not replace)
2. Figure out why CronTriggerTimetable does not work with the `else` branch and fix that.

Reply:
This logic was added in #24262; CronTriggerTimetable was added in #25503 roughly a month later.

Reply:
1. Included both cron timetables in the if statement

2. `next_dagrun_info` was always the same [here](https://github.com/apache/airflow/blob/main/airflow/www/views.py#L2844). So we never reached either of the `break` statements.

Reply:
Hmm, that does not sound right, the timetable should always be progressing forward. @mai-nakagawa maybe you could spot what’s wrong here?

(In the mean time I’m going to add a check to prevent infinite loop from happening)

Reply:
Ah I get it, because CronTriggerTimetable always uses the current time when catchup=False without considering `last_automated_data_interval`. It should; I’ll fix it in a different PR.

Reply:
@uranusjr Does your PR mean we don't need this PR at all?

Reply:
This would still serve as an optimisation.